Dockerfile

```diff
@@ -1,6 +1,8 @@
 # syntax = docker/dockerfile:latest
 
-FROM alpine:3.18.3
+ARG ALPINE_VERSION=3.18
+
+FROM alpine:$ALPINE_VERSION
```

Comment:
LGTM. 

The only potential downside is that the minor version (instead of patch) doesn't get a "release" as often per se, so we'd have to have a think about how we're doing the docker builds. 

Maybe monthly? Or pending a release of something (e.g. pin `jq` and `curl` versions)?

The other *potential* issue, is that by not changing the version, the docker cache may not be invalidated where in this particular case, we wish it to be.

There are ways of getting around this moderately easily, but it's something to consider if we're going down this route. Maybe if we go down the package pinning route instead of Alpine version pinning, this may be a better solution.

Alpine docker is a strange beast when it comes to updates for sure. 

So that I'm not only naysayer holding this up, I'm going to say it's good to go, but with the caveat that we should probably investigate how we're going to do things in future.

Reply:
LGTM. 

The only potential downside is that the minor version (instead of patch) doesn't get a "release" as often per se, so we'd have to have a think about how we're doing the docker builds. 

Maybe monthly? Or pending a release of something (e.g. pin `jq` and `curl` versions)?

The other *potential* issue, is that by not changing the version, the docker cache may not be invalidated where in this particular case, we wish it to be.

There are ways of getting around this moderately easily, but it's something to consider if we're going down this route. Maybe if we go down the package pinning route instead of Alpine version pinning, this may be a better solution.

Alpine docker is a strange beast when it comes to updates for sure. 

So that I'm not only naysayer holding this up, I'm going to say it's good to go, but with the caveat that we should probably investigate how we're going to do things in future. I've used edge for my own projects for quite a while now without issues (https://github.com/modem7/Dnscrypt-Proxy/blob/master/Dockerfile), so if everyone is happy with that, I'd also be happy (we'd pin packages instead of Alpine versions).

Reply:
@modem7 I'll make a little refactoring of github action and about the concern about building cache it's addressable.

About versioning, I can ask alpine team about this.

I'll aprove and defer a new build now.